include/hobbes/eval/probes.H

```diff
@@ -0,0 +1,29 @@
+
+#ifndef HOBBES_EVAL_PROBES_HPP_INCLUDED
+#define HOBBES_EVAL_PROBES_HPP_INCLUDED
+
+#include <cstdint>
+#include <string>
+
+namespace hobbes {
+
+class cc;
+
+void initProbeDefs(cc&);
+
+struct InjectedProbeMetadata {
+  size_t id;
+  std::string name;
+  std::string file;
+  size_t line;
+  std::string function;
+  void* callSite;
+  void* returnSite;
+
+  static constexpr uint64_t typeTag = static_cast<uint64_t>(0x01) << 56;
+  static constexpr size_t caveSize = 5;
```

Comment:
Why 5?  Sorry if I've missed something obvious.

Reply:
I'm inserting a 5 byte no-op in the injectProbe intrinsic because that's the default cave size Xpedite appears to use (for a jmp instruction I guess) and I wanted something concrete. You could probably let the injectProbe call specify size provided you can map size -> multibyte no-op correctly though.